api/auth/cas_auth.py

```diff
@@ -127,6 +127,8 @@ def validate_bearer(token):
     current_app.logger.debug("invalid bearer token")
     return None
 
+def validate_third_party(secret_token):
+    return secret_token == settings.THIRD_PARTY_SECRET_TOKEN
```

Comment:
Same here regarding the assumption of just one 3rd party. 
```suggestion
    return secret_token == settings.THIRD_PARTY_SECRET_TOKEN_GITLAB
```